src/Plant/hydraulics/get_struct_p_end_from_q.jl

```diff
@@ -0,0 +1,105 @@
+# this function returns the end pressure for a root struct
+# flow in mol s^-1, p_end in MPa
+function get_struct_p_end_from_q(any_struct::StructTreeRootLayer, flow::FT; p_ini::FT=FT(Inf)) where {FT}
+    if p_ini==Inf
+        p_end = any_struct.p_ups
+    else
+        p_end = p_ini
+    end
+
+    #=
+    compute the pressure drop along the rhizosphere, note that p_end is corrected for minor change in surface tension
+    1. Using van Genuchten equation to compute soil hydraulic conductance Krhiz in the rhizosphere
+    2. Calculate dP = flow / Krhiz_i for each of the 10 shells of the rhizosphere
+    3. Update the P at the end of the rhizosphere
+    =#
+    a   = any_struct.soil_a
+    m   = any_struct.soil_m
+    n   = any_struct.soil_n
+    k   = any_struct.k_rhiz
+    k_s = get_relative_surface_tension(any_struct.t_soil)
+    k_t = get_relative_viscosity(any_struct.t_soil)
+    for i in 0:9
+        if p_end<0
+            shell_t = (1.0 / (1.0 + (a*(-p_end/k_s))^n)) ^ m
+        else
+            shell_t = 1.0
+        end
```

Comment:
Add `FT` for type stability, also, can we incorporate these hard-codes numbers into a model?